dali/test/dali_operator_test.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TEST_DALI_OPERATOR_TEST_H_
+#define DALI_TEST_DALI_OPERATOR_TEST_H_
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <memory>
+#include <dali/pipeline/pipeline.h>
+#include <dali/test/op_graph.h>
+#include <dali/test/tensor_list_wrapper.h>
+
+namespace dali {
+namespace testing {
+
+using Arguments = std::map<std::string, double>; // TODO: some generalization. boost::any?
+
+
+
+class DaliOperatorTest : public ::testing::Test, public ::testing::WithParamInterface<Arguments> {
+public:
+  using Verify = std::function<void(TensorListWrapper, TensorListWrapper, Arguments)>;
+protected:
+  void RunTest(TensorListWrapper inputs, TensorListWrapper outputs, Arguments operator_arguments, Verify verify) {
+  }
+
+
+private:
+  virtual std::unique_ptr<OpDag> GenerateOperatorsGraph() const noexcept = 0;
```

Comment:
We're no better returning it though `unique_ptr` than by value - the creator has to relinquish ownership of the object anyway. As such, just returning it with all the benefits of move semantics seems like a better idea (less boilerplate code on the implementor's side).

Reply:
I did it just to stub `OpDag` without hacky downcasting, but yeah, I guess returning by value is better here